lib/document.js

```diff
@@ -1258,6 +1258,29 @@ Document.prototype.$__registerHooks = function () {
 
   DocumentArray || (DocumentArray = require('./types/documentarray'));
 
+  function emitErrorOnModel(err) {
+    // emit on the Model if listening
+    if (this.constructor.listeners('error').length) {
+      this.constructor.emit('error', err);
+    } else {
+      // emit on the connection
+      if (!this.db.listeners('error').length) {
+        err.stack = 'No listeners detected, throwing. '
+                  + 'Consider adding an error listener to your connection.\n'
+                  + err.stack
+      }
+      this.db.emit('error', err);
+    }
+  }
+
+  // Don't duplicate calls to validate() on sub-documents
+  // validate already descends the object tree on its own
+  if(! this.__parent) {
```

Comment:
Then I don't think this is going to work in 3.8.x unfortunately. However, master passes the test you specified :)